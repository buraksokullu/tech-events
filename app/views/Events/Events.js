import React, { Component } from 'react';
import PropTypes from 'prop-types';

import Button from 'Components/FormFields/Button/Button';
import ClockIcon from 'Components/svg/ClockIcon';
import WorldIcon from 'Components/svg/WorldIcon';
import { formatDate, getTime, getTimeDifference } from 'Utils/helper';

import s from './Events.scss';

export class Events extends Component {
  getCityName = id => {
    const { cities } = this.props;

    if (cities !== null) {
      return cities.find(x => x.id === id).name;
    }
    return '';
  };

  render() {
    const { events, toggleModal, cancelEvent } = this.props;
    return (
      <div>
        <div className={s.dashboard}>
          {events && events.length > 0 ? (
            events.map(item => (
              <div className={s.event} key={item.id}>
                <div className={s.date}>{formatDate(item.startDate)}</div>
                <div className={s.item}>
                  <div className={s.firstRow}>
                    <div className={s.eventName}>
                      {item.isFree && <span className={s.isFree}> FREE </span>}
                      <strong className={s.name}>{item.name}</strong>
                    </div>
                    {item.isSelected ? (
                      <Button onClick={() => cancelEvent(item.id)}> Cancel </Button>
                    ) : (
                      <Button onClick={() => toggleModal(item.id, item.name)}> Sign Up </Button>
                    )}
                  </div>
                  <div className={s.secondRow}>
                    <span className={s.cityName}>
                      <WorldIcon />
                      <span>{this.getCityName(item.city)}</span>
                    </span>
                    <span className={s.timeDifferences}>
                      {getTimeDifference(item.startDate, item.endDate)}
                    </span>
                    <div className={s.fromToEvent}>
                      <ClockIcon />
                      <span className={s.time}>{getTime(item.startDate, item.endDate)}</span>
                    </div>
                  </div>
                </div>
              </div>
            ))
          ) : (
            <div>No events found :(</div>
          )}
        </div>
      </div>
    );
  }
}

Events.propTypes = {
  events: PropTypes.arrayOf(PropTypes.shape()),
  cities: PropTypes.arrayOf(PropTypes.shape()),
  toggleModal: PropTypes.func.isRequired,
  cancelEvent: PropTypes.func.isRequired
};

Events.defaultProps = {
  events: [],
  cities: []
};

export default Events;
