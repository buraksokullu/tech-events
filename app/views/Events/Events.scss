.dashboard {
  .event {
    .date {
      color: gray;
      margin-bottom: 10px;
    }

    .item {
      display: flex;
      flex-direction: column;
      width: 100%;
      padding: 20px;
      border: 1px solid #e5e5e5;
      box-shadow: 0 4px 8px 0 rgba(0, 0, 0, 0.16);
      border-radius: 8px;
      margin-bottom: 30px;

      &:hover {
        border: solid 1px rgba(0, 0, 0, 0.08);
      }

      .firstRow {
        display: flex;
        align-items: center;
        justify-content: space-between;
        margin-bottom: 30px;
        .eventName {
          display: flex;
          align-items: center;
          flex-direction: row;

          .isFree {
            display: flex;
            justify-content: center;
            align-items: center;
            height: 18px;
            line-height: 1px;
            margin-right: 20px;
            padding: 8px;
            font-size: 12px;
            color: white;
            background: $greenlight60;
            border-radius: 10px;
          }

          .name {
            font-size: 16px;
            @media screen and (max-width: 767px) {
              font-size: 14px;
              max-width: 160px;
            }
          }
        }
      }

      .secondRow {
        display: flex;
        align-items: center;
        justify-content: space-between;

        .cityName {
          display: flex;
          align-items: center;
          width: 200px;
          color: $greenlight70;

          @media screen and (max-width: 767px) {
            width: 105px;
          }

          span{
            margin-left: 5px;
          }
        }

        .timeDifferences {
          color: $grey60;
        }

        .fromToEvent {
          display: flex;
          align-items: center;
          .time {
            margin-left: 5px;
            width: 155px;
            font-weight: 600;
            color: $grey60;

            @media screen and (max-width: 767px) {
              width: auto;
              font-size:13px;
            }
          }
        }
      }
    }
  }
}
