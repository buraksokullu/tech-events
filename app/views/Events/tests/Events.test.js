import React from 'react';
import { shallow } from 'enzyme';
import Events from '../Events';

describe('Events', () => {
  let eventsProps = null;

  beforeEach(() => {
    eventsProps = {
      cities: null
    };
  });
  it('Should render correctly', () => {
    const component = shallow(<Events {...eventsProps} />);

    expect(component).toMatchSnapshot();
  });
  it('getCityName should return names while city array is not null', () => {
    const eventsProps = {
      cities: [
        {
          id: 1,
          name: 'Barcelona'
        }
      ]
    };
    const component = shallow(<Events {...eventsProps} />);
    const instance = component.instance();

    expect(instance.getCityName(1)).toEqual('Barcelona');
  });
  it('getCityName should return empty while city array is null ', () => {
    const component = shallow(<Events {...eventsProps} />);
    const instance = component.instance();

    expect(instance.getCityName(1)).toEqual('');
  });
});
