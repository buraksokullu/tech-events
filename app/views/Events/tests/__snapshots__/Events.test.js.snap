// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Events Should render correctly 1`] = `
<div>
  <div>
    <div>
      No events found :(
    </div>
  </div>
</div>
`;
